File: app/src/main/java/com/nakharin/wongfah/network/model/JsonMenu.java
package com.nakharin.wongfah.network.model;

import com.google.gson.annotations.SerializedName;

import org.parceler.Parcel;

@Parcel
public class JsonMenu {

    JsonMenu() {
    }

    @SerializedName("id")
    public int id = -1;

    @SerializedName("name")
    public String name = "";

    @SerializedName("image_url")
    public String imageUrl = "";

    @SerializedName("price")
    public int price = 0;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }
}

File: app/src/main/java/com/nakharin/wongfah/network/model/JsonMenu.kt
package com.nakharin.wongfah.network.model

import android.os.Parcel
import com.google.gson.annotations.SerializedName
import com.nakharin.wongfah.utility.KParcelable
import com.nakharin.wongfah.utility.parcelableCreator

class JsonMenu() : KParcelable {

    @SerializedName("id")
    var id = -1

    @SerializedName("name")
    var name = ""

    @SerializedName("image_url")
    var imageUrl = ""

    @SerializedName("price")
    var price = 0

    constructor(p: Parcel) : this() {
        id = p.readInt()
        name = p.readString()
        imageUrl = p.readString()
        price = p.readInt()
    }

    override fun writeToParcel(dest: android.os.Parcel, flags: Int) {
        dest.writeInt(id)
        dest.writeString(name)
        dest.writeString(imageUrl)
        dest.writeInt(price)
    }

    companion object {
        @JvmField val CREATOR = parcelableCreator(::JsonMenu)
    }
}
